datafusion/functions/src/string/common.rs

```diff
@@ -68,6 +69,74 @@ pub(crate) fn general_trim<T: OffsetSizeTrait>(
         },
     };
 
+    if use_string_view {
+        string_view_trim::<T>(trim_type, func, args)
+    } else {
+        string_trim::<T>(trim_type, func, args)
+    }
+}
+
+// removing 'a will cause compiler complaining lifetime of `func`
+fn string_view_trim<'a, T: OffsetSizeTrait>(
```

Comment:
This implementation is an improvement -- however, it will now copy the string values to a new `StringArray`

I think if the function generated StringView output directly it would be possible to avoid all string copies (and only adjust the views).

I can file this as a follow on optimization idea.

However, we can